restapi/src/main/java/io/stargate/web/docsapi/resources/DocumentResourceV2.java

```diff
@@ -308,6 +311,74 @@ public Response putDoc(
         });
   }
 
+  @POST
+  @ManagedAsync
+  @ApiOperation(
+      value =
+          "Execute a built-in function (e.g. $push and $pop) against this document. Performance may vary.")
+  @ApiResponses(
+      value = {
+        @ApiResponse(code = 200, message = "OK", response = WriteDocResponse.class),
+        @ApiResponse(code = 400, message = "Bad request", response = Error.class),
+        @ApiResponse(code = 401, message = "Unauthorized", response = Error.class),
+        @ApiResponse(code = 403, message = "Forbidden", response = Error.class),
+        @ApiResponse(code = 422, message = "Unprocessable entity", response = Error.class),
+        @ApiResponse(code = 500, message = "Internal Server Error", response = Error.class)
+      })
+  @Path("collections/{collection-id}/{document-id}/{document-path: .*}/function")
+  @Consumes({MediaType.APPLICATION_JSON})
+  @Produces(MediaType.APPLICATION_JSON)
+  public Response executeBuiltInFunction(
+      @Context HttpHeaders headers,
+      @Context UriInfo ui,
+      @ApiParam(
+              value =
+                  "The token returned from the authorization endpoint. Use this token in each request.",
+              required = true)
+          @HeaderParam("X-Cassandra-Token")
+          String authToken,
+      @ApiParam(value = "the namespace that the collection is in", required = true)
+          @PathParam("namespace-id")
+          String namespace,
+      @ApiParam(value = "the name of the collection", required = true) @PathParam("collection-id")
+          String collection,
+      @ApiParam(value = "the name of the document", required = true) @PathParam("document-id")
+          String id,
+      @ApiParam(
+              value = "the path in the JSON that you want to execute the operation on",
+              required = true)
+          @PathParam("document-path")
+          List<PathSegment> path,
+      @ApiParam(value = "The operation to perform", required = true) @Valid
+          ExecuteBuiltInFunction payload,
+      @ApiParam(
+              value = "Whether to include profiling information in the response (advanced)",
+              defaultValue = "false")
+          @QueryParam("profile")
+          Boolean profile,
+      @Context HttpServletRequest request) {
+    return handle(
+        () -> {
+          ExecutionContext context = ExecutionContext.create(profile);
+          JsonNode result =
+              documentService.executeBuiltInFunction(
```

Comment:
should we try to make this reactive? we kind of wanted to transit to the reactive way of processing?